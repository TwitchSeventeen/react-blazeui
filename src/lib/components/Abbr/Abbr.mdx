---
name: Abbr
---

import { Playground, PropsTable } from 'docz'
import Abbr from './index';

# Abbr

Used to mark up an abbreviation for accessibility. It can be used inline (ie it's not a block level element).

## Properties
<PropsTable of={Abbr} />

## Basic usage
<Playground>
    <Abbr title="help text for accessibility">
        abbreviation
    </Abbr>
</Playground>