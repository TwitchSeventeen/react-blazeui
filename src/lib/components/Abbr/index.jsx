import React from 'react';
import PropTypes from 'prop-types';

function Abbr (props) {
  const { children, title } = props;
  return (
      <abbr
          className="u-text--help"
          title={title}
      >
        { children }
      </abbr>
  );
}

Abbr.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Title attribute for the abbreviation component
   */
  title: PropTypes.string.isRequired,
};

Abbr.defaultProps = {
  children: null,
};

export default Abbr;