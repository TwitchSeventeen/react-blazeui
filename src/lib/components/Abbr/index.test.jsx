import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Abbr from './index';

describe('Abbr', () => {
  it('should be defined', () => {
    expect(Abbr).toBeDefined();
  });

  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
        <Abbr title="Help text">
          lorem ipsum dolor
        </Abbr>
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
