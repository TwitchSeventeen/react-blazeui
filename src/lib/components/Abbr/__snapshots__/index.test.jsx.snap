// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Abbr renders correctly 1`] = `
<abbr
  className="u-text--help"
  title="Help text"
>
  lorem ipsum dolor
</abbr>
`;
